src/utils/pickers/pickers.ts

```diff
@@ -31,15 +32,35 @@ export async function projectPicker(
       displayScanId: undefined,
       scanDatetime: undefined
     });
-    updateState(context, constants.branchIdKey, { id: undefined, name: constants.branchLabel, displayScanId: undefined, scanDatetime: undefined});
-    updateState(context, constants.scanIdKey, { id: undefined, name: constants.scanLabel, displayScanId: undefined, scanDatetime: undefined});
-
+    updateState(context, constants.branchIdKey, { id: undefined, name: constants.branchLabel, displayScanId: undefined, scanDatetime: undefined });
+    updateState(context, constants.scanIdKey, { id: undefined, name: constants.scanLabel, displayScanId: undefined, scanDatetime: undefined });
+    await setDefaultBranch(item.id, context, logs);
     await vscode.commands.executeCommand(commands.refreshTree);
     quickPick.hide();
   });
   quickPick.show();
 }
 
+async function setDefaultBranch(
+  projectId: string,
+  context: vscode.ExtensionContext,
+  logs: Logs,
+) {
+  try {
+    const gitApi = await getGitAPIRepository();
+    const gitBranchName = gitApi.repositories[0]?.state.HEAD?.name; // TODO: replace with getFromState(context, constants.branchName) when the onBranchChange is working properly
```

Comment:
Lines 50-51 appear more than once in the entire PR, mabye try to use a utility function like :
async function getCurrentGitBranch() {
const gitApi = await getGitAPIRepository();
return gitApi.repositories[0]?.state.HEAD?.name;
}